R/preprocess.R

```diff
@@ -0,0 +1,69 @@
+#' Add columns for interval censoring of primary and secondary events
+#'
+#' @param linelist ...
+#' @param ptime_lwr ...
+#' @param ptime_upr ...
+#' @param pwindow ...
+#' @param stime_lwr ...
+#' @param stime_upr ...
+#' @param swindow ...
+#' @family preprocess
+#' @autoglobal
+#' @export
+add_event_vars <- function(
+  linelist, ptime_lwr = NULL, ptime_upr = NULL, pwindow = NULL,
```

Comment:
I think adding event vars is really going from having p time in some format and s time in some format (lower bound assumed) and adding pwindow and window to get the event variables. 

I think the step of making sure we have _lwr and upr for each is another function (and would perhaps naturally live in a general as_epidist_linelist class which can then optionally be passed to as_latent_model. 

If not that s3 design then something else.